lib/push.py

```diff
@@ -114,6 +114,32 @@ def prepare(self):
                                      "digest": "sha256:" + hash_c })
       # Prepare metadata.
       ch.INFO("preparing metadata")
+      self.image.metadata_load()
+      hist = self.image.metadata["history"]
+      # Some registries, e.g., Quay, use history metadata for simple sanity
+      # checks. For example, when an image's number of "empty_layer" history
+      # entries doesn't match the number of layers being uploaded, Quay will
+      # reject the image upload.
+      #
+      # This type of error checking is odd as the empty_layer key is optional
+      # (https://github.com/opencontainers/image-spec/blob/main/config.md).
+      #
+      # Thus, to push images built (or pulled) with Charliecloud we ensure the
+      # the total number of non-empty layers always totals one (1). To do this
+      # we iterate over the history entires backward searching for the first
+      # non-empty entry and preserve it; all others are set to empty.
+      non_empty_winner = None
+      for i in range(len(hist) - 1, -1, -1):
+         if (   "empty_layer" not in hist[i].keys()
+             or (    "empty_layer" in hist[i].keys()
+                 and hist[i]["empty_layer"] == 'false')):
```

Comment:
Is it really a string? Below it's a boolean.

```suggestion
                 and not hist[i]["empty_layer"])):
```